language_detector/main.py

```diff
@@ -1,8 +1,9 @@
-# -*- coding: utf-8 -*-
-
-"""This is the entry point of the program."""
-
 def detect_language(text, languages):
     """Returns the detected language of given text."""
-    # implement your solution here
-    pass
+    counts = { lang['name']:0 for lang in languages }
+    words = text.split()
+    for w in words:
+        for lang in languages:
+            if w in lang['common_words']:
+                counts[lang['name']]+=1
+    return max((count, name) for name, count in counts.items())[1]
```

Comment:
Yeah, I saw and liked that one too. The way it works is that the key parameter lets you customize how things like max, sort, etc do the comparison. And when just iterating over a dict, it defaults to iterating over the keys.

so `max(counts, key = counts.get)` basically goes through each of the keys in counts, ie, each of the names of the languages... but doesn't compare them directly. Instead it compares, the value of, counts.get(name) to determine which is larger (where 'name' represents the keys/language names) (`counts.get(name)` would basically be equivalent to `counts[name]`)